hudi-common/src/main/java/org/apache/hudi/metrics/MetricsReporterFactory.java

```diff
@@ -66,28 +64,29 @@ public static Option<MetricsReporter> createReporter(HoodieWriteConfig config, M
 
     switch (type) {
       case GRAPHITE:
-        reporter = new MetricsGraphiteReporter(config, registry);
+        reporter = new MetricsGraphiteReporter(metricsConfig, registry);
         break;
       case INMEMORY:
         reporter = new InMemoryMetricsReporter();
         break;
       case JMX:
-        reporter = new JmxMetricsReporter(config, registry);
+        reporter = new JmxMetricsReporter(metricsConfig, registry);
         break;
       case DATADOG:
-        reporter = new DatadogMetricsReporter(config, registry);
+        reporter = new DatadogMetricsReporter(metricsConfig, registry);
         break;
       case PROMETHEUS_PUSHGATEWAY:
-        reporter = new PushGatewayMetricsReporter(config, registry);
+        reporter = new PushGatewayMetricsReporter(metricsConfig, registry);
         break;
       case PROMETHEUS:
-        reporter = new PrometheusReporter(config, registry);
+        reporter = new PrometheusReporter(metricsConfig, registry);
         break;
       case CONSOLE:
         reporter = new ConsoleMetricsReporter(registry);
         break;
       case CLOUDWATCH:
-        reporter = new CloudWatchMetricsReporter(config, registry);
+        reporter = (MetricsReporter) ReflectionUtils.loadClass("org.apache.hudi.metrics.cloudwatch.CloudWatchMetricsReporter",
```

Comment:
why we could not access CloudWatchMetricsReporter?
we are changing the arg in the constructor from write config to metricsconfig right. 
should we not just use the new constructor

Reply:
CloudWatchMetricsReporter is part of hudi-aws. It could not be moved to hudi-common so using Reflection here

Reply:
Added a test for reflection as well